src/main/java/org/spongepowered/api/item/inventory/Container.java

```diff
@@ -57,17 +57,24 @@
      *
      * @param viewer The viewer to show this inventory to
      * @param cause The {@link Cause} to use when opening the inventory
-     * @throws IllegalArgumentException if a {@link PluginContainer} is not the root of the cause
+     * @return {@code true}, if the {@link Container} was opened, {@code false}
+     *         otherwise.
+     * @throws IllegalArgumentException if a {@link PluginContainer} is not the
+     *         root of the cause
      */
-    void open(Player viewer, Cause cause) throws IllegalArgumentException;;
+    boolean open(Player viewer, Cause cause) throws IllegalArgumentException;
 
     /**
-     * Stops showing this Inventory to the given viewer.
+     * Stops showing this Inventory to the given viewer, if it is currently
+     * being shown.
      *
      * @param viewer The viewer to stop showing this inventory to
      * @param cause The {@link Cause} to provide when closing the inventory
-     * @throws IllegalArgumentException if a {@link PluginContainer} is not the root of the cause
+     * @return {@code true}, if the {@link Container} was closed, {@code false}
+     *         otherwise.
+     * @throws IllegalArgumentException if a {@link PluginContainer} is not the
```

Comment:
If the cause is removed, the API methods should probably not throw an IllegalArgumentException any more right?